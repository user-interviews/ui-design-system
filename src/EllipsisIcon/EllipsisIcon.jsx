import React from 'react';
import PropTypes from 'prop-types';

import classNames from 'classnames';

import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faEllipsisV } from '@fortawesome/free-solid-svg-icons';

const EllipsisIcon = ({ className }) => (
  <FontAwesomeIcon
    className={classNames(className)}
    icon={faEllipsisV}
  />
);

EllipsisIcon.propTypes = {
  className: PropTypes.string,
};

EllipsisIcon.defaultProps = {
  className: undefined,
};

export default EllipsisIcon;
