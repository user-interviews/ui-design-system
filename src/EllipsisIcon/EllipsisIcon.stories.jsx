import React from 'react';

import EllipsisIcon from 'src/EllipsisIcon';

export default {
  title: 'Design System/EllipsisIcon',
  component: EllipsisIcon,
};

export const Default = () => (
  <EllipsisIcon />
);
