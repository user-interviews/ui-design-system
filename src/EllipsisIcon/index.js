import EllipsisIcon from './EllipsisIcon';

export default EllipsisIcon;
