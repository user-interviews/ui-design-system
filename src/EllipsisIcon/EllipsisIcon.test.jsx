import React from 'react';

import { render } from '@testing-library/react';

import EllipsisIcon from './EllipsisIcon';

describe('EllipsisIcon', () => {
  it('renders the expected snapshot', () => {
    const { container } = render(<EllipsisIcon />);
    expect(container.firstChild).toMatchSnapshot();
  });

  describe('with a className set', () => {
    it('renders the expected snapshot', () => {
      const { container } = render(<EllipsisIcon className="hello-world" />);
      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
