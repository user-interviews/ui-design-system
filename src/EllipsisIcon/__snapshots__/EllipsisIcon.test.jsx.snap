// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EllipsisIcon renders the expected snapshot 1`] = `
<svg
  aria-hidden="true"
  class="svg-inline--fa fa-ellipsis-v fa-w-6 "
  data-icon="ellipsis-v"
  data-prefix="fas"
  focusable="false"
  role="img"
  viewBox="0 0 192 512"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
    fill="currentColor"
  />
</svg>
`;

exports[`EllipsisIcon with a className set renders the expected snapshot 1`] = `
<svg
  aria-hidden="true"
  class="svg-inline--fa fa-ellipsis-v fa-w-6 hello-world"
  data-icon="ellipsis-v"
  data-prefix="fas"
  focusable="false"
  role="img"
  viewBox="0 0 192 512"
  xmlns="http://www.w3.org/2000/svg"
>
  <path
    d="M96 184c39.8 0 72 32.2 72 72s-32.2 72-72 72-72-32.2-72-72 32.2-72 72-72zM24 80c0 39.8 32.2 72 72 72s72-32.2 72-72S135.8 8 96 8 24 40.2 24 80zm0 352c0 39.8 32.2 72 72 72s72-32.2 72-72-32.2-72-72-72-72 32.2-72 72z"
    fill="currentColor"
  />
</svg>
`;
